src/main/java/org/jenkinsci/plugins/fodupload/DastScanSharedBuildStep.java

```diff
@@ -280,6 +291,236 @@ public void saveReleaseSettingsForWorkflowDrivenScan(String userSelectedRelease,
+            throws IllegalArgumentException, IOException {
+
+        DastScanController dynamicController = new DastScanController(getApiConnection(), null, Utils.createCorrelationId());
+        try {
+
+            PutDastAutomatedPostmanReqModel dastPostmanScanSetupReqModel;
+            dastPostmanScanSetupReqModel = new PutDastAutomatedPostmanReqModel();
+            dastPostmanScanSetupReqModel.setEntitlementFrequencyType(entitlementFreq);
+            dastPostmanScanSetupReqModel.setAssessmentTypeId(Integer.parseInt(assessmentTypeID));
+            dastPostmanScanSetupReqModel.setTimeZone(timeZone);
+            dastPostmanScanSetupReqModel.setEntitlementId(Integer.parseInt(entitlementId));
+            dastPostmanScanSetupReqModel.setDynamicScanEnvironmentFacingType(scanEnvironment);
+
+
+            if (requireNetworkAuth) {
+                PutDastScanSetupReqModel.NetworkAuthentication networkAuthentication = dastPostmanScanSetupReqModel.getNetworkAuthenticationSettings();
+                networkAuthentication.setPassword(networkAuthPassword);
+                networkAuthentication.setUserName(networkAuthUserName);
+
+                if (!networkAuthType.isEmpty()) {
+                    networkAuthentication.setNetworkAuthenticationType((networkAuthType));
+                } else
+                    throw new IllegalArgumentException("Network Auth Type not set for releaseId: " + userSelectedRelease);
+                networkAuthentication.setRequiresNetworkAuthentication(true);
+                dastPostmanScanSetupReqModel.setNetworkAuthenticationSettings(networkAuthentication);
+            }
+            if (postmanIdCollection == null) {
+                throw new IllegalArgumentException(String.format("Postman Scan - one of the id is  not set for release Id={%s}"
+                        , userSelectedRelease));
+            }
+            else {
+                    dastPostmanScanSetupReqModel.setCollectionFileIds(ConvertStringtoIntArr(postmanIdCollection));
+                }
+
+            PutDastScanSetupResponse response = dynamicController.putDastPostmanScanSettings(Integer.parseInt(userSelectedRelease),
+                    dastPostmanScanSetupReqModel);
+            if (response.isSuccess && response.errors == null) {
+                System.out.println("Successfully saved settings for release id = " + userSelectedRelease);
+
+            } else {
+                throw new Exception(String.format("Failed to save scan settings for release id %d", Integer.parseInt(userSelectedRelease)));
+            }
+        } catch (Exception e) {
+            throw new IllegalArgumentException(String.format("Failed to save scan settings for release id %d", Integer.parseInt(userSelectedRelease)), e);
+        }
+    }
+    public int[] ConvertStringtoIntArr(String fileIds) {
+        String[] postmanIds = fileIds.split(",");
+        int[] postmanIdArr = new int[postmanIds.length];
```

Comment:
Should we check for the null here before accessing the length property

Reply:
This is controller part. I belive all checks inclusing this should happen before reaching to it. Should we do it again here.?